crunch-core/src/main/java/org/apache/crunch/io/FormatBundle.java

```diff
@@ -290,7 +291,21 @@ public void readFields(DataInput in) throws IOException {
       String value = Text.readString(in);
       extraConf.put(key, value);
     }
-    if (in.readBoolean()) {
+    boolean hasFilesystem;
+    try {
+      hasFilesystem = in.readBoolean();
+    } catch (EOFException e) {
+      // This can be a normal occurrence when Crunch is treated as a  cluster-provided
+      // dependency and the version is upgraded.  Some jobs will have been submitted with
+      // code that does not contain the filesystem field.  If those jobs run later with
+      // this code that does contain the field, EOFException will occur trying to read
+      // the non-existent field.
+      LOG.debug("EOFException caught attempting to read filesystem field.  This condition"
+          + "may temporarily occur with jobs that are submitted before but run after a"
```

Comment:
```suggestion
      LOG.debug("EOFException caught attempting to read filesystem field.  This condition "
          + "may temporarily occur with jobs that are submitted before but run after a "
```

Reply:
Patch LGTM w/the whitespace fix included; assuming no objections, I'll merge it later today.